### ISSUE TITLE
CNN Text classification validation result is unstable in training process

### ISSUE THREAD

**Issue by rtygbwwwerr** [keras-team/keras#8185]
![cnn-training](https://user-images.githubusercontent.com/8509559/31753483-71d6e296-b4c3-11e7-8c4c-1f38a80623a2.png)

I using the  up-to-date Keras for my Text classification, the training data set has 1 million items while the valid set has 10000 items, and the summary info of training model is:
Layer (type)                 Output Shape              Param #   
=================================================================
merge_1 (Merge)              (None, 2560)              0         
_________________________________________________________________
dense_1 (Dense)              (None, 128)               327808    
_________________________________________________________________
dropout_4 (Dropout)          (None, 128)               0         
_________________________________________________________________
dense_2 (Dense)              (None, 256)               33024     
_________________________________________________________________
dropout_5 (Dropout)          (None, 256)               0         
_________________________________________________________________
dense_3 (Dense)              (None, 3)                 771       
=================================================================
Total params: 477,197
Trainable params: 476,141
Non-trainable params: 1,056
_________________________________________________________________
merge layer is concat from three network:
merged = Merge([model1, model2, model3], mode='concat')
each sub model stacked by 3 cnn layers.

In training process, the acc in training set is steadily increasing while on the valid set is unstable(please check the attach picture). I use mini-batch for training(batch size = 256), and randomly shuffle the data in each batch. Who can tell me what's wrong with my model?
